app/controllers/rover_shifts_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class RoverShiftsController < ApplicationController
+  load_and_authorize_resource :rover_shift
+
+  def index
+    @job_id = params[:job_id].presence
+
+    @rover_shifts = @job_id ? @rover_shifts.where(job_id: @job_id) : RoverShift.none
+    return unless @job_id
+
+    @activity_executions = ActivityExecution.ordered_by_rover_shift_prio.joins(:activity)
```

Comment:
```suggestion
    @activity_executions = ActivityExecution.ordered_by_rover_shift_prio
```

Already joined in scope